src/java/voldemort/serialization/SerializerDefinition.java

```diff
@@ -139,11 +138,38 @@ public boolean equals(Object obj) {
             return false;
         SerializerDefinition s = (SerializerDefinition) obj;
         return Objects.equal(getName(), s.getName())
-               && Objects.equal(this.schemaInfoByVersion, s.schemaInfoByVersion)
                && Objects.equal(this.compression, s.compression)
                && this.hasVersion == s.hasVersion();
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        boolean isEqual = commonEquals(obj);
+        if(!isEqual) {
+            return false;
+        }
+        SerializerDefinition s = (SerializerDefinition) obj;
+        return Objects.equal(this.schemaInfoByVersion, s.schemaInfoByVersion);
+    }
+
+    public boolean semanticEquals(Object obj) {
+        boolean isEqual = commonEquals(obj);
```

Comment:
I don't understand why would you need ```commonEquals``` here?

What you're doing in the beginning of the ```semanticEquals``` is exactly the same thing that you are doing in the ```equals()``` function.

I think you can just leave the ```equals()``` as it was before, and depend on it from your new ```semanticEquals()``` function.